.github/workflows/macos-universal.yml

```diff
@@ -0,0 +1,207 @@
+name: macOS universal OpenSSL+libssh
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ devel ]
+  pull_request:
+
+env:
+  OPENSSL_VER: "3.1.4"
+  LIBSSH_VER: "0.11.2"
+
+jobs:
+  macos-arm64-build:
+    name: Build (arm64)
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Prep dirs
+        run: |
+          mkdir -p build arm64
+          echo "ROOT=$PWD" >> $GITHUB_ENV
+
+      - name: Build OpenSSL (arm64)
+        run: |
+          set -euxo pipefail
+          cd build
+          curl -LO "https://www.openssl.org/source/openssl-${OPENSSL_VER}.tar.gz"
+          tar xzf "openssl-${OPENSSL_VER}.tar.gz"
+          cd "openssl-${OPENSSL_VER}"
+          ./Configure darwin64-arm64-cc --prefix="$GITHUB_WORKSPACE/arm64/openssl" --libdir=lib \
+            no-tests no-ssl3 no-weak-ssl-ciphers enable-ec_nistp_64_gcc_128
+          make -j"$(sysctl -n hw.ncpu)"
```

Comment:
It's best to split separate invocations across separate steps in the job definition. Otherwise, it'd difficult to scan the logs and sometimes outputs of multiple things can be confused with each other.